gcsio/src/main/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageGrpcReadChannel.java

```diff
@@ -104,7 +104,7 @@ public class GoogleCloudStorageGrpcReadChannel implements SeekableByteChannel {
   private boolean channelIsOpen = true;
 
   // Current position in the object.
-  private long positionInGrpcStream;
+  private long positionInGrpcStream = -1;
```

Comment:
Wouldn't this cause the index to go out of sync since at many places we are using +=.

eg: https://github.com/GoogleCloudDataproc/hadoop-connectors/blob/master/gcsio/src/main/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageGrpcReadChannel.java#L530

It might be better to have another variable to track this rather than overloading this variable. Single Responsibility  i.e.

Reply:
The reason I used this is because of same reason. This should tell us the positionInGrpcStream. Until we open the grpcStream position is undefined (denoted by -1). Once we open it, this variable is set to correct position. Similar implementation is present in json implementation.

Reply:
The sequential coupling here is the one that worries me. Can we add some asserts where we expect this to be initialized?

Reply:
Sure - let me do that.